services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -468,86 +521,185 @@ async def get_template_project(
+                    changed_entries = {}
+                    for node_key, new_node_data in new_partial_workbench_data.items():
+                        current_node_data = project.get("workbench", {}).get(node_key)
+
+                        if current_node_data is None:
+                            log.debug(
+                                "node %s is missing from project, no patch", node_key
+                            )
+                            raise NodeNotFoundError(project_uuid, node_key)
+                        # find changed keys
+                        changed_entries.update(
+                            {
+                                node_key: _find_changed_dict_keys(
+                                    current_node_data,
+                                    new_node_data,
+                                    look_for_removed_keys=False,
+                                )
+                            }
+                        )
+                        # patch
+                        current_node_data.update(new_node_data)
+                    return (project, changed_entries)
+
+                new_project_data, changed_entries = _patch_workbench(
+                    current_project, partial_workbench_data
+                )
+
+                # update timestamps
+                new_project_data["lastChangeDate"] = now_str()
+
+                log.debug("DB updating with %s", pformat(new_project_data))
+                result = await conn.execute(
+                    # pylint: disable=no-value-for-parameter
+                    projects.update()
+                    .values(**_convert_to_db_names(new_project_data))
+                    .where(projects.c.id == current_project[projects.c.id.key])
+                    .returning(literal_column("*"))
+                )
+                project: RowProxy = await result.fetchone()
+                log.debug("DB updated returned %s", pformat(project))
+                user_email = await self._get_user_email(conn, project.prj_owner)
+
+                tags = await self._get_tags_by_project(
+                    conn, project_id=project[projects.c.id]
+                )
+                return (
+                    _convert_to_schema_names(project, user_email, tags=tags),
+                    changed_entries,
+                )
+
+    async def replace_user_project(
         self,
-        project_data: Dict,
+        new_project_data: Dict[str, Any],
         user_id: int,
         project_uuid: str,
         include_templates: Optional[bool] = False,
-    ) -> Dict:
-        """updates a project from a user"""
+    ) -> Dict[str, Any]:
+        """updates a project from a user
+        replace_project is True then new_project_data shall contain a full valid project.
+        if replace_project is False then new_project_data shall contain only the modified elements.
+        """
```

Comment:
I see no `replace_project` in this function. What am I missing?

Reply:
good one. I changed the method too many times and the comments got outdated.